megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -2157,6 +2170,172 @@ private void launchFighters(MovePath path) {
+        // Get conventional infantry if manual mode is available
+        List<Entity> enemyInfantry = new ArrayList<>();
+        if (game.getOptions().booleanOption(OptionsConstants.ADVCOMBAT_TACOPS_MANUAL_AMS)) {
+            for (Entity curEnemy : this.getEnemyEntities().stream().filter(Entity::isDeployed).collect(Collectors.toSet())) {
+                if (curEnemy.getPosition() != null && curEnemy.isVisibleToEnemy()) {
+
+                    if (curEnemy instanceof Infantry && !(curEnemy instanceof EjectedCrew)) {
+                        enemyInfantry.add(curEnemy);
+                    }
+
+                }
+            }
+        }
+
+        for (Entity curEntity : this.getEntitiesOwned()) {
+            if (!curEntity.isDeployed() || curEntity.getPosition() == null) {
+                continue;
+            }
+
+            List<WeaponMounted> activeAMS = curEntity.
+                    getWeaponList().
+                    stream().
+                    filter(w -> w.getType().hasFlag(AmmoWeapon.F_AMS) && w.hasModes()).
+                    collect(Collectors.toList());
+
+            if (!activeAMS.isEmpty()) {
+
+                // Set default to on/automatic and test to see if it should be off or manual instead
+                EquipmentMode newAMSMode = EquipmentMode.getMode(Weapon.MODE_AMS_ON);
+
+                boolean isOverheating = (curEntity instanceof Mech) && (curEntity.getHeat() >= 14);
+
+                // If there are enough nearby enemy infantry (only counted if the game option is
+                // set), choose manual fire
+                if (!enemyInfantry.isEmpty() && !curEntity.isAirborne()) {
+                    int infantryRange = enemyInfantry.stream().mapToInt(e -> Compute.effectiveDistance(game, curEntity, e)).min().getAsInt();
+                    if (infantryRange <= 3) {
+                        newAMSMode = EquipmentMode.getMode(Weapon.MODE_AMS_MANUAL);
+                    }
+                }
+
+                // If AMS was used manually this round, chances are it will be needed next round too
+                if (usedManualAMS(curEntity.getId())) {
+                    newAMSMode = EquipmentMode.getMode(Weapon.MODE_AMS_MANUAL);
+                }
+
+                for (WeaponMounted curAMS : activeAMS) {
+
+                    EquipmentMode curMode = curAMS.curMode();
+
+                    // Turn off laser AMS to help with overheating problems
+                    if (curAMS.getType().hasFlag(WeaponType.F_ENERGY)) {
+                        if (isOverheating) {
+                            newAMSMode = EquipmentMode.getMode(Weapon.MODE_AMS_OFF);
+                        }
+                    } else {
+
+                        // Determine if ammo needs to be conserved
+                        boolean conserveAmmo = curAMS.getLinkedAmmo().getUsableShotsLeft() <= (int) Math.floor(curAMS.getOriginalShots() *
+                                behaviorSettings.getSelfPreservationValue() / 100.0);
```

Comment:
Could this be wrapped into the existing ammo conservation code?

Reply:
Maybe, but there's a distinct difference.  The existing ammo conservation code is a shoot/no shoot determination.  For AMS it is actually a mode change that prevents automatic operation in the following turn.  Keeps the existing code about determining whether to fire, and this on whether to turn this specific system off or on.